src/test/utils/Wallet.sol

```diff
@@ -16,6 +16,14 @@ contract Wallet is ERC721Holder, ERC1155Holder {
         IERC20(token).transfer(to, amount);
     }
 
+    function setAllowance20(
```

Comment:
let's be consistent with the naming. looking at the function below it's opt'd with `transferERC721` so this should be `setAllowanceERC20`

Reply:
we can also consider consolidate them into 1 function and use introspection to smartly decide which function to call.

Reply:
Here again I think a generalized function which uses ERC165 smartly doesn't make sense, since the three ERCs are conceptually different e.g. need to specify an amount when setting allowance in ERC20 vs setting approval in ERC721

Reply:
yup, agree.